tests: adds test case for incorrect pattern validation.

jcchavezs opened the pull request:
This PR test case shows a test case where a string with pattern `[a-z0-9]{32,}` is matched by `7f46165474d11ee5836777d85df2cdab$` and not by `7$f46165474d11ee5836777d85df2cdab`.

Ping @marc-gr

johandorland commented:
* The test suite is copy pasted from https://github.com/json-schema-org/JSON-Schema-Test-Suite. If you want to add test cases you should create a PR there because I'm not going to maintain separate test cases from theirs in the JSON format.
* The test you added is actually incorrect. The pattern is anchored, meaning the string should end with an alphanumeric string of length 32, neither of which the test cases do.

jcchavezs commented:
Hi @johandorland,

> The test suite is copy pasted from https://github.com/json-schema-org/JSON-Schema-Test-Suite. If you want to add test cases you should create a PR there because I'm not going to maintain separate test cases from theirs in the JSON format.

Sorry I did not notice that.

> The test you added is actually incorrect. The pattern is anchored, meaning the string should end with an alphanumeric string of length 32, neither of which the test cases do.

I fixed the test. As I updated in the description, `7f46165474d11ee5836777d85df2cdab$` matches  the pattern `[a-z0-9]{32,}` unless I am missing something.

johandorland commented:
I'm still closing this PR due to the first bullet point.
